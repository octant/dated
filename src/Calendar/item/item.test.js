/* eslint-env jest */

import React from "react";
import Item from "./";
import renderer from "react-test-renderer";
import serializer from "jest-glamor-react";
import Enzyme, { mount } from "enzyme";
import Adaptor from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adaptor() });
expect.addSnapshotSerializer(serializer);

describe("<Item /> appearance", () => {
  test("renders the same each time", () => {
    const component = renderer.create(<Item label="3" />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Item /> functionality", () => {
  test("displays label property", () => {
    const text = "Mo";
    const item = mount(<Item label={text} />);
    expect(item.text()).toEqual(text);
  });
});
