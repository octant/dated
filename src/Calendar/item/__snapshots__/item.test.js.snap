// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> appearance renders the same each time 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  box-sizing: border-box;
  float: left;
  margin: 0em;
  text-align: center;
  background-color: rgba(0, 0, 0, 0);
  color: white;
  width: undefinedundefined;
  height: undefinedundefined;
  line-height: undefinedundefined;
}

<div
  className="css-0"
  onClick={[Function]}
>
  3
</div>
`;
